File: MetinTarama/MetinTarama.java
package metintarama;

import java.util.Scanner;

public class MetinTarama {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Tarama yapmak istediğiniz metni girin: ");
        String metin = reader.nextLine();
        System.out.print("Arama yapmak istediğiniz kelimeyi girin: ");
        String kelime = reader.nextLine();
        int parca = kelime.length();
        int tekrar = metin.length();
        int tekrarSayisi = 0;
        for (int i = 0; i < tekrar; i++) {
            String kontrol = metin.substring(i, i + parca);
            if (kontrol.equals(kelime)) {
                tekrarSayisi += 1;
                System.out.println("Aranan kelime " + (i + 1) + ". harften başlıyor");
            }
        }
        System.out.println("Tüm metinde "+tekrarSayisi+" defa geçmektedir.");
    }
}

File: MetinTarama/MetinTarama.py
def metin_tarama(hedef_metin, aranan_kelime):

    toplam_gecme_sayisi = 0

    for i in range(len(hedef_metin) - len(aranan_kelime) + 1):
        if hedef_metin[i:i+len(aranan_kelime)].lower() == aranan_kelime.lower():
            print("{} - {} indexleri arasında geçmekte".format(i, i+len(aranan_kelime)))
            toplam_gecme_sayisi += 1

    print("{} aranan metin içinde {} defa geçmektedir.".format(aranan_kelime, toplam_gecme_sayisi))


metin_tarama(input("Tarama yapılacak metni girin : "), input("Aranan kelimeyi girin : "))